File: src/bots/mcts.js
// MCTS bot — Monte Carlo Tree Search with UCB1. Phase C.
import { validCols, checkWin, isFull } from '../game/rules.js';
import { clone, place } from '../game/board.js';
import { AI, HUMAN } from '../game/constants.js';

/**
 * Return the best column for `token` using MCTS.
 * @param {number[][]} board
 * @param {number} token  - AI or HUMAN
 * @param {object} opts
 * @param {number} opts.iterations  - number of simulations (default 1000)
 * @param {number} opts.C           - UCB1 exploration constant (default √2)
 * @returns {number} column index
 */
export function mctsBot(board, token, { iterations = 1000, C = Math.SQRT2 } = {}) {
  // TODO: implement selection, expansion, simulation, backpropagation
}

File: src/bots/mcts.py
# MCTS bot — Monte Carlo Tree Search with UCB1. Phase C stub.
import math
from ..game.rules import valid_cols, check_win, is_full
from ..game.board import clone, place
from ..game.constants import AI, HUMAN


def mcts_bot(board, token, iterations=1000, C=math.sqrt(2)):
    """
    Return the best column for `token` using MCTS.

    board:      current board state
    token:      AI or HUMAN
    iterations: number of simulations (default 1000)
    C:          UCB1 exploration constant (default √2)
    Returns:    column index
    """
    # TODO: implement selection, expansion, simulation, backpropagation
    pass
